cmd/kubelet/app/options/docker.go

```diff
@@ -0,0 +1,81 @@
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package options
+
+import (
+	"runtime"
+
+	"github.com/spf13/pflag"
+)
+
+const (
+	// When these values are updated, also update test/e2e/framework/util.go
+	defaultPodSandboxImageName    = "gcr.io/google_containers/pause"
+	defaultPodSandboxImageVersion = "3.0"
+	defaultPodSandboxImage        = defaultPodSandboxImageName +
+		"-" + runtime.GOARCH + ":" +
+		defaultPodSandboxImageVersion
+)
+
+type DockerOptions struct {
+	// PodSandboxImage is the image whose network/ipc namespaces
+	// containers in each pod will use.
+	PodSandboxImage string
+	// DockerEndpoint is the path to the docker endpoint to communicate with.
+	DockerEndpoint string
+	// networkPluginName is the name of the network plugin to be invoked for
+	// various events in kubelet/pod lifecycle
+	NetworkPluginName string
```

Comment:
In the new CRI world, network plugins are container runtime details. kubelet is not aware of them.

The tricky part is that these are still consumed by rkt since they haven't switched to CRI. That's why I marked this PR as WIP.